backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/model/validator/M_Product.java

```diff
@@ -58,4 +68,31 @@ public void updateAllM_HU_PI_Item_Products(final I_M_Product product)
 			InterfaceWrapperHelper.save(huPIP);
 		}
 	}
+
+	@ModelChange(timings = ModelValidator.TYPE_BEFORE_CHANGE, ifColumnsChanged = I_M_Product.COLUMNNAME_IsActive)
+	public void isActive(@NonNull final I_M_Product product)
+	{
+		if (product.isActive())
+		{
+			return;
+		}
+
+		final IStorageEngine storageEngine = storageEngineProvider.getStorageEngine();
+
+		final IStorageQuery storageQuery = storageEngine
+				.newStorageQuery()
+				.addProductId(ProductId.ofRepoId(product.getM_Product_ID()))
+				.setExcludeAfterPickingLocator(false)
+				.setOnlyStockedProducts(true)
+				.setOnlyActiveHUs(true)
+				.setIgnoreBPartnerColumn();
+
+		final IContextAware context = PlainContextAware.createUsingOutOfTransaction();
+		final boolean isEmptyStock = storageEngine.retrieveStorageRecords(context, storageQuery).isEmpty();
```

Comment:
why not introducing storageEngine.anyMatching(storageQuery) method?

i.e. why shall we load all records when we just want to check if there is any matching.

Reply:
will do, thx